helixbass/tsc-rust
# Parse + bind interface declaration

In this PR:
- parse and bind a simple interface declaration like:
```
interface Foo {
  bar: number
}
```

To test:
If you run `cargo run tmp.tsx` against a source file whose contents are:
```
interface Foo {
  bar: number
}
```
then you should see debug-logging of a `SourceFile` whose only statement is an `InterfaceDeclaration` with `name: "Foo"` and `members:` contains a `PropertySignature` with `name: "bar"` and `type_:` a `KeywordTypeNode` with `kind: NumberKeyword`. And in the `post-binding:` debug-logging you should see that the `locals` symbol table on the `SourceFile` contains a `Symbol` for `"Foo"`, which itself has a nested `members` symbol table containing a `Symbol` for `"bar"`. Then it should panic during type-checking

Based on `array-literal-declaration`
